### ISSUE TITLE
[BUG] Stuck on "Ensuring login on Bing"

### ISSUE THREAD

**Issue by Real-Zac** [charlesbel/Microsoft-Rewards-Farmer#131]
It is stuck on "[LOGIN] Ensuring login on Bing..."

(https://user-images.githubusercontent.com/118482504/202889436-bab86a1d-5e5a-436b-9317-19e31dfa8343.png)

All login information is correct and there is proper syntax in the accounts file.

**Comment by inverspe** [comment]
restart

**Comment by DumbAsFuck** [comment]
what i did was schedule the script to run in the background 6 times a day at semi random intervals. has been working very well for me.

**Comment by semiino** [comment]
I figured out that the script works properly only if it logs on to Bing with the original set language (for example I originally set the account using Bing in French. When I try to log in using Bing in another language it won't show me my points etc. and the script stays stuck on "Ensuring login on Bing...").
Therefore I changed the code on lines 91 and 133 using:

browser.get('https://www.bing.com/?setlang=fr&cc=fr&cc=FR')

instead of: 

browser.get('https://www.bing.com/')

It worked for me

**Comment by inverspe** [comment]
> I figured out that the script works properly only if it logs on to Bing with the original set language (for example I originally set the account using Bing in French. When I try to log in using Bing in another language it won't show me my points etc. and the script stays stuck on "Ensuring login on Bing..."). Therefore I changed the code on lines 91 and 133 using:
> 
> browser.get('https://www.bing.com/?setlang=fr&cc=fr&cc=FR') //This example is for the French version, you need to change the URL according to the language you need
> 
> instead of:
> 
> browser.get('https://www.bing.com/')
> 
> It worked for me

does the mobile search work for you? mine doesnt

**Comment by semiino** [comment]
> > I figured out that the script works properly only if it logs on to Bing with the original set language (for example I originally set the account using Bing in French. When I try to log in using Bing in another language it won't show me my points etc. and the script stays stuck on "Ensuring login on Bing..."). Therefore I changed the code on lines 91 and 133 using:
> > browser.get('https://www.bing.com/?setlang=fr&cc=fr&cc=FR') //This example is for the French version, you need to change the URL according to the language you need
> > instead of:
> > browser.get('https://www.bing.com/')
> > It worked for me
> 
> does the mobile search work for you? mine doesnt

Yes, it does.

Alternatively, you can edit line 221:

return('fr-FR', 'FR', '120')
//Instead of fr (French) write down the abbreviation of the language you want to set.


I hope this helps!

**Comment by Ector70** [comment]
I've tried various solutions but it still doesn't work

**Comment by semiino** [comment]
> 

Are you using the headless mode? If so try not to use it by commenting lines 32 and 33, if you can watch what the program does it should be easier to figure out what's going on.

**Comment by inverspe** [comment]
> > 
> 
> Are you using the headless mode? If so try not to use it by commenting lines 32 and 33, if you can watch what the program does it should be easier to figure out what's going on.

when mine starts auto mobile search it gets stuck on ensuring login

**Comment by inverspe** [comment]
> > > I figured out that the script works properly only if it logs on to Bing with the original set language (for example I originally set the account using Bing in French. When I try to log in using Bing in another language it won't show me my points etc. and the script stays stuck on "Ensuring login on Bing..."). Therefore I changed the code on lines 91 and 133 using:
> > > browser.get('https://www.bing.com/?setlang=fr&cc=fr&cc=FR') //This example is for the French version, you need to change the URL according to the language you need
> > > instead of:
> > > browser.get('https://www.bing.com/')
> > > It worked for me
> > 
> > 
> > does the mobile search work for you? mine doesnt
> 
> Yes, it does.
> 
> Alternatively, you can edit line 221:
> 
> return('fr-FR', 'FR', '120') //Instead of fr (French) write down the abbreviation of the language you want to set.
> 
> I hope this helps!

what chrome version are you using?

**Comment by inverspe** [comment]
I FINALLY FIXXED IT BY REMOVING LINE 86 

checkBingLogin(browser, isMobile)

**Comment by inverspe** [comment]
guys remove line 86 to fix mobile search

checkBingLogin(browser, isMobile)

**Comment by AlexyGalaxy** [comment]
> 

This does fix the issue, however now POINTS_COUNTER = 0 and the script will always print "You have 0 points on your account!"

Do you know how to fix this?
![0 points](https://user-images.githubusercontent.com/86795638/211838119-9a5d95df-c7da-4fa4-b80c-0536587b6b1f.JPG)

**Comment by AlexyGalaxy** [comment]
> guys remove line 86 to fix mobile search
> 
> checkBingLogin(browser, isMobile)

DO NOT COMMENT OUT LINE 86 if you do POINTS_COUNTER will not work properly.

Instead try this:
-How to Fix Mobile:
Comment out around line 134-138 with double quotes above and below
""
#Refresh page
browser.get('https://bing.com/')
#Wait 5 seconds
time.sleep(10)
#Update Counter
""

-How to run Script:

**Comment by LamoosaCaboosa** [comment]
none of these fixes worked for me... is there a working fix?

**Comment by inverspe** [comment]
> none of these fixes worked for me... is there a working fix?

same

**Comment by LamoosaCaboosa** [comment]
<img width="465" alt="Screen Shot 2023-01-16 at 5 38 52 PM" src="https://user-images.githubusercontent.com/64758730/212775474-58462901-4076-48a1-a8a6-0fca2948a439.png">

not sure what this does, but now it all works perfectly fine for me! :) used most of the tips here as well as this change and its just working now

**Comment by LamoosaCaboosa** [comment]
i guess its just skipping mobile, but if thats the cost, im fine with it

**Comment by inverspe** [comment]
> i guess its just skipping mobile, but if thats the cost, im fine with it

You fixed mobile search?

**Comment by LamoosaCaboosa** [comment]
nooooonono, i just turned it off until i find a way to make it work again

**Comment by inverspe** [comment]
> nooooonono, i just turned it off until i find a way to make it work again

Ah okay I did the same i removed the checkmobile thing and works fine again but it won’t check how much points you have

**Comment by LamoosaCaboosa** [comment]
The way that I showed still counts the points just fine. I'm still trying to figure out how to make mobile work

**Comment by inverspe** [comment]
> The way that I showed still counts the points just fine. I'm still trying to figure out how to make mobile work

Yes you remove line 86 if you want mobile search to work again don’t change line 87

**Comment by LamoosaCaboosa** [comment]
<img width="875" alt="Screen Shot 2023-01-16 at 7 57 57 PM" src="https://user-images.githubusercontent.com/64758730/212787207-f09a4410-086d-4fda-8e47-5b8607703f33.png">
(for my method for turning off mobile mode you need to change this a little too)

**Comment by AlexyGalaxy** [comment]
![try this](https://user-images.githubusercontent.com/86795638/212793734-f7ab5424-c7c8-4db3-b496-fdc0992e6026.JPG)

Like i said before, do not comment out line 87 it brakes the code in ways you do not understand.
Instead comment out these 4 lines of code. This made mobile work for me.

**Comment by AlexyGalaxy** [comment]
Since yall dont understand. I forked this project and uploaded my current working version that will complete all searches and punch cards. 

https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

**Comment by inverspe** [comment]
> Since yall dont understand. I forked this project and uploaded my current working version that will complete all searches and punch cards. Also made the README even more simple to understand.
> 
> https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

yeah mobile search works now but another problem arose now it takes roughly around 10 minutes to complete a single mobile search

**Comment by AlexyGalaxy** [comment]
You mean it stays on 1/20 for 10 mins or it takes 10 mins to complete all 20 searches?

Although not ideal, everything WILL complete. The longer time the scrip takes the less likely to get banned from microsoft. This is completely against their TOS.

**Comment by AlexyGalaxy** [comment]
One of my accounts is stuck on mobile seach 7/20 for over an hour. I think I know how to fix it. I will make an edit soon and update my code on github.

**Comment by LamoosaCaboosa** [comment]
<img width="875" alt="Screen Shot 2023-01-17 at 6 59 15 PM" src="https://user-images.githubusercontent.com/64758730/213038363-c37a128f-00ed-40cf-ad4c-4c19abb03b74.png">
this happened when i tried! :( but your version definetely shows better errors.

**Comment by AlexyGalaxy** [comment]
I am close to fixing the issues. Currently everything works except for mobile searches. Only way i get mobile searches to run is by commenting out the lines below. I am looking for a soultion to implement. I will update my github now.

line 1022
```
completeDailySet(browser)
```
line 1028
```
#completePunchCards(browser)
```
line 1034
```
#completeMorePromotions(browser)
```

**Comment by AlexyGalaxy** [comment]
> <img alt="Screen Shot 2023-01-17 at 6 59 15 PM" width="875" src="https://user-images.githubusercontent.com/64758730/213038363-c37a128f-00ed-40cf-ad4c-4c19abb03b74.png">
> 
> this happened when i tried! :( but your version definetely shows better errors.

I just updated my github. It now completes all pc and mobile searches without issues. 

If you have any more issues let me know.

**Comment by AlexyGalaxy** [comment]
I just updated the code again. Everything seems to be working now. Again if you have issues let me know.

https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

If all goes well next update should include Account.Logger.txt and Error.Logger.txt

**Comment by LamoosaCaboosa** [comment]
It has been working wonderfully for me so far! Great work!

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
